src/Generator/InvoiceGenerator.php

```diff
@@ -81,7 +87,10 @@ public function generateForOrder(OrderInterface $order, \DateTimeInterface $date
             $order->getCurrencyCode(),
             $order->getLocaleCode(),
             $order->getTotal(),
-            $this->lineItemsConverter->convert($order),
+            new ArrayCollection(array_merge(
+                $this->orderItemUnitsToLineItemsConverter->convert($order)->toArray(),
+                $this->shippingAdjustmentsToLineItemsConverter->convert($order)->toArray()
```

Comment:
As far as I see, in `LitemItemsConverter`s we return `ArrayCollection`s to convert them to arrays here, merge and convert to the one `ArrayCollection` one more time 😄 Is there any reason why `OrderItemUnitsToLineItemsConverter` and `ShippingAdjustmentsToLineItemsConverter` cannot return a simple array?

Reply:
To be honest, I didn't want to change the return types of these methods, but I'm not aware of any reason for that in first place